softwerkskammer/test/mailsender/mailsenderService_test.js

```diff
@@ -250,14 +259,153 @@ describe("MailsenderService", () => {
       const sentMail = singleSentEmail();
       expect(sentMail.bcc).to.contain("memberA");
       expect(sentMail.html).to.contain("mark down");
       expect(sentMail.icalEvent).to.not.contain("BEGIN:VCALENDAR");
       expect(sentMail.icalEvent).to.not.contain("URL:http://localhost:17125/activities/urlOfTheActivity");
       expect(statusmessage.contents().type).to.equal("alert-success");
     });
+
+    describe("sending in chunks", () => {
+      let membersAboveSingleChunkThreshhold = [
+        new Member({ email: "memberA" }),
+        new Member({ email: "memberB" }),
+        new Member({ email: "memberC" }),
+        new Member({ email: "memberD" }),
+        new Member({ email: "memberE" }),
+        new Member({ email: "memberF" }),
+      ];
+
+      it("sends in chunks to members of selected groups when above configured threshold", async () => {
+        sinon.stub(groupsAndMembersService, "addMembersToGroup").callsFake((group) => {
+          group.members = membersAboveSingleChunkThreshhold;
+        });
+        const statusmessage = await mailsenderService.sendMailToInvitedGroups(["GroupA"], undefined, message, sender);
+        const allSent = allSentEmail();
+        const sentMail1 = allSent[0];
+        expect(sentMail1.bcc).to.eql(["memberA", "memberB", "memberC", "memberD", "memberE"]);
+        const sentMail2 = allSent[1];
+        expect(sentMail2.bcc).to.eql(["memberF"]);
+
+        expect(statusmessage.contents().type).to.equal("alert-success");
+      });
+
+      it("sends status mail to intiator after all mails have been sent", async () => {
+        let mailsSentCount = 0;
+        let myResolve;
+        const allExpectedMailsSent = new Promise((resolve) => {
+          myResolve = resolve;
+        });
+        sendmail.reset();
+        sendmail.callsFake(() => {
+          mailsSentCount++;
+
+          if (mailsSentCount >= 3) {
+            myResolve(undefined);
+          }
+        });
+        sinon.stub(groupsAndMembersService, "addMembersToGroup").callsFake((group) => {
+          group.members = membersAboveSingleChunkThreshhold;
+          return group;
+        });
+
+        let statusmessage = await mailsenderService.sendMailToInvitedGroups(["GroupA"], undefined, message, sender);
+
+        expect(statusmessage.contents().type).to.equal("alert-success");
+        expect(statusmessage.contents().text).to.equal("message.content.mailsender.success");
+
+        await allExpectedMailsSent;
+        const allSent = allSentEmail();
+
+        expect(allSent).to.have.length(3);
+
+        expect(allSent[2].to).to.equal(sender.email());
+        expect(allSent[2].subject).to.contain("Report");
+        expect(allSent[2].subject).to.contain(mailSubject);
+        expect(allSent[2].html).to.contain("erfolgreich");
+      });
+
+      it("returns success regardless of mail sending result", async () => {
```

Comment:
Das find ich nicht gut. Status in erfolgsmail soll status von meldung sein

Reply:
Bei einen Tests hat aber die Ersetzung der i18n-Keys nicht stattgefunden (und die Frage ist ja auch welches locale man dann verwenden soll bzw. wie man das durschleift?).

Reply:
Was ich meine: die Mails werden nicht durch die Übersetzung gejagt, zumindest sah das bei mir so aus (und ist ja auch irgendwie logisch, man was ja nicht welche Sprache die richtige ist), aber eventuell sprechen wir auch von zwei verschiedenen Fragestellungen?